tasks/eva_3399/src/main/groovy/eva3399/RemediateIndels.groovy

```diff
@@ -0,0 +1,249 @@
+            def mergeOpId = "EVA3399_MERGED_${this.assembly}_${mergee.accession}_${mergeTarget.accession}".toString()
+            def mergeOpClass = (mergee.accession >= 5e9) ? svoeClass : dbsnpSvoeClass
+            def mergeOp = new SubmittedVariantOperationEntity()
+            mergeOp.fill(EventType.MERGED, mergee.accession, mergeTarget.accession,
+                    "EVA3399 - Hash collision with another SS after normalization",
+                    Arrays.asList(new SubmittedVariantInactiveEntity(mergee)))
+            mergeOp.setId(mergeOpId)
+            this.prodEnv.mongoTemplate.save(mergeOp, this.prodEnv.mongoTemplate.getCollectionName(mergeOpClass))
+        }
+    }
+
+    def _updateSSWithRS = {Map<Long, SubmittedVariantEntity> svesGroupedByRsHash, Map<String, Long> rsIDsByHash ->
+        rsIDsByHash.each {rsHash, rsID ->
+            def sves = svesGroupedByRsHash.get(rsHash)
+            def svesHash = sves.collect{it.hashedMessage}
+            sves.each {it.setClusteredVariantAccession(rsID)}
+            sves.each{logger.info("Updating SS ${it.accession} with RS ${rsID}...")}
+            [sveClass, dbsnpSveClass].each {collectionClass ->
+                this.prodEnv.mongoTemplate.updateMulti(query(where("_id").in(svesHash)),
+                        Update.update("rs", rsID), collectionClass)
+            }
+        }
+    }
+
+    def _clearExistingRs = {List<SubmittedVariantEntity> sves ->
+        def svesHash = sves.collect{it.hashedMessage}
+        sves.each{it.setClusteredVariantAccession(null)}
+        [sveClass, dbsnpSveClass].each {ssClass ->
+            this.prodEnv.mongoTemplate.updateMulti(query(where("_id").in(svesHash)), new Update().unset("rs"), ssClass)
+        }
+    }
+
+    def _assignNewRs = {List<SubmittedVariantEntity> svesToInsert ->
+        def svesGroupedByRsHash = svesToInsert.groupBy { EVAObjectModelUtils.getClusteredVariantHash(it)}
+        def rsRecordsToCreate = svesToInsert.collect { EVAObjectModelUtils.toClusteredVariant(it)}
+        def rsIDsByHash = this.prodEnv.clusteredVariantAccessioningService.getOrCreate(rsRecordsToCreate).collectEntries { [it.hash, it.accession] }
+        _updateSSWithRS(svesGroupedByRsHash, rsIDsByHash)
+    }
+
+    def _assignRs = {List<SubmittedVariantEntity> svesToInsert ->
+        _clearExistingRs(svesToInsert)
+        def svesGroupedByRsHash = svesToInsert.groupBy { EVAObjectModelUtils.getClusteredVariantHash(it)}
+        def existingRsIDsByHash = [cveClass, dbsnpCveClass].collect{collectionClass ->
+            this.prodEnv.mongoTemplate.find(query(where("_id")
+                    .in(svesGroupedByRsHash.keySet())), collectionClass)
+        }.flatten().collectEntries{[it.hashedMessage, it.accession]}
+        _updateSSWithRS(svesGroupedByRsHash, existingRsIDsByHash)
+        if (this.targetAssembly) {
+            _assignNewRs(svesGroupedByRsHash.findAll { rsHash, _ ->
```

Comment:
What I think this is saying is that if we're not working with a current target assembly, we'll still cluster the SS that hash to an existing RS, but not create new RS... is that correct?  If so that's different from what we normally do, which is not assign any RS to SS in non-target assemblies.  (It's not necessarily a problem, just want to be sure I'm understanding.)

Reply:
Now addressed...